### ISSUE TITLE
no-unused-vars suggests breaking change.

### ISSUE THREAD

**Issue by Summertime** [denoland/deno_lint#1251]
### Lint Name

no-unused-vars

### Code Snippet

```ts
const obj = { unwanted: 1, wanted: 2 };
const { unwanted, ...result } = obj;
console.log(result);
```

### Expected Result

Either no error, or tells me to replace it with `unwanted: _unwanted`

### Actual Result

> If this is intentional, prefix it with an underscore like `_unwanted`

### Additional Info

changing it to just `_unwanted` changes how the program runs

#687 looks potentially similar, but seems to be different enough that this might be best as a different issue.

### Version

deno 1.40.5 (release, x86_64-unknown-linux-gnu)
v8 12.1.285.27
typescript 5.3.3
